### PR TITLE
Feat/pupil 731/quiz counter visual

### PR DESCRIPTION
**Pull request by kimon-satan**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

Added visual indicator to the quiz counter

## Link to the design doc

https://www.figma.com/design/YcWQMMhHPVVmc47cHHEEAl/%F0%9F%9F%A2-Oak-Design-Kit?m=auto

## A link to the component in the deployment preview

https://deploy-preview-233--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-pupil-oakquizcounter--docs

## Testing instructions

try changing the number of counter in the story

## ACs

- [x]  Updated OakQuizCounter to include a visual indicator
- [x]  screen and tablet views
- [x]  Handles maximum number of questions (demoed in story)

### CODE REVIEW

**Review by messmike — APPROVED** [review]

File: src/components/organisms/pupil/OakQuizCounter/OakQuizCounter.tsx [hunk 1]
```diff
@@ -1,23 +1,49 @@
...
+    >
+      <OakSpan $font={"heading-light-4"} $color={"text-disabled"}>
+        <OakSpan $font={"heading-4"} $color={"text-primary"}>
+          {counter}{" "}
+        </OakSpan>
+        of {total}
       </OakSpan>
-      of {total}
-    </OakSpan>
+      <OakFlex $gap={"space-between-ssx"}>{pills}</OakFlex>
```
Comment (BTitterington) [inline src/components/organisms/pupil/OakQuizCounter/OakQuizCounter.tsx:46]: $flexWrap="wrap" will improve mobile view when there are too many to display.

Comment (kimon-satan) [reply src/components/organisms/pupil/OakQuizCounter/OakQuizCounter.tsx:46]: > $flexWrap="wrap" will improve mobile view when there are too many to display.

I'll add it but it actually doesn't render on mobile

Comment (BTitterington) [reply src/components/organisms/pupil/OakQuizCounter/OakQuizCounter.tsx:46]: Ahh, fair enough.

Comment (RBrownsett) [reply src/components/organisms/pupil/OakQuizCounter/OakQuizCounter.tsx]: What is the expected behaviour on mobile? After around 9 questions the lines do not show in storybook

Comment (kimon-satan) [reply src/components/organisms/pupil/OakQuizCounter/OakQuizCounter.tsx]: > What is the expected behaviour on mobile? After around 9 questions the lines do not show in storybook

It doesn't render ion mobile instead a different component is used

**Comment by RBrownsett** [comment]
@kimon-satan getting a contrast issue on the grey text - maybe needs to a shade darker?

image

**Comment by kimon-satan** [comment]
> @kimon-satan getting a contrast issue on the grey text - maybe needs to a shade darker?
> 
> image

Nice catch but that's one for Mike and also another ticket. The text colour hasn't been changed from what is now in prod

**Comment by RBrownsett** [comment]
> > @kimon-satan getting a contrast issue on the grey text - maybe needs to a shade darker?
> > image
> 
> Nice catch but that's one for Mike and also another ticket. The text colour hasn't been changed from what is now in prod

Ah, I'll get a ticket raised and check with @messmike

**Comment by kimon-satan** [comment]
> > > @kimon-satan getting a contrast issue on the grey text - maybe needs to a shade darker?
> > > image
> > 
> > 
> > Nice catch but that's one for Mike and also another ticket. The text colour hasn't been changed from what is now in prod
> 
> Ah, I'll get a ticket raised and check with @messmike

cheers. So can I have your blessing for this one ?

**Review by RBrownsett — APPROVED** [review]
Approved - ticket will be raised for contrast issue

**Comment by messmike** [comment]
Good spot @RBrownsett - text-subdued, #575757 will pass contrast

**Pull request merged**